format/spec.md

```diff
@@ -57,6 +57,7 @@ In addition to row-level deletes, version 2 makes some requirements stricter for
 
 ## Overview
 
+<!-- markdown-link-check-disable-next-line -->
```

Comment:
should this link maybe have `https://iceberg.apache.org/` as its prefix as it's already being used in other places for images?